File: Controllers/TestArrayList/Person.cs
using System.Diagnostics;
using System.Text.Json;
using FilmesApi.TestArrayList;


class Person
{
    private Dictionary<string, dynamic> arGithubEventPrepared;
    private List<GithubEvent> arGithubEvent;

    public Person(string id_pessoa)
    {
        Stopwatch stopwatch = new Stopwatch();
        stopwatch.Start();

        LoadData();
        // PrepareData();
        FindPerson(id_pessoa);

        stopwatch.Stop();
        TimeSpan elapsed = stopwatch.Elapsed;
        Console.WriteLine("Total time [benchmark]: " + elapsed.Milliseconds + " ms");
    }

    private void LoadData()
    {
        Stopwatch stopwatch = new Stopwatch();
        stopwatch.Start();

        using (FileStream stream = File.OpenRead("Controllers/TextArrayList/large-file.json"))
        {
            using (StreamReader reader = new StreamReader(stream))
            {
                string json = reader.ReadToEnd();
                arGithubEvent = JsonSerializer.Deserialize<List<GithubEvent>>(json);
            }
        }

        stopwatch.Stop();
        TimeSpan elapsed = stopwatch.Elapsed;
        Console.WriteLine("LoadData [benchmark]: " + elapsed.Milliseconds + " ms");
    }

    // private void LoadData()
    // {
    //     Stopwatch stopwatch = new Stopwatch();
    //     stopwatch.Start();
    //     string json = File.ReadAllText("large-file.json");
    //     arGithubEvent = JsonSerializer.Deserialize<List<dynamic>>(json);
    //     // string json = File.ReadAllText("large-file.json");
    //     // arGithubEvent = JsonConvert.DeserializeObject<List<dynamic>>(json);
    //     stopwatch.Stop();
    //     TimeSpan elapsed = stopwatch.Elapsed;
    //     Console.WriteLine("LoadData [benchmark]: " + elapsed.Milliseconds + " ms");
    // }

    private void PrepareData()
    {
        Stopwatch stopwatch = new Stopwatch();
        stopwatch.Start();

        arGithubEventPrepared = new Dictionary<string, dynamic>();
        foreach (dynamic githubEvent in arGithubEvent)
        {
            if (githubEvent.TryGetProperty("id", out JsonElement idElement))
            {
                string id = idElement.GetString();
                arGithubEventPrepared.Add(id, githubEvent);

            }
        }

        stopwatch.Stop();
        TimeSpan elapsed = stopwatch.Elapsed;
        Console.WriteLine("PrepareData [benchmark]: " + elapsed.Milliseconds + " ms");
    }

    private void FindPerson(string id_pessoa)
    {
        Stopwatch stopwatch = new Stopwatch();
        stopwatch.Start();

        GithubEvent foundEvent = arGithubEvent.FirstOrDefault(eventObj => eventObj.Id == id_pessoa);

        if (foundEvent != null)
        {
            // Console.WriteLine("Event found: " + foundEvent.Id);
        }
        else
        {
            Console.WriteLine("Event not found with id: " + id_pessoa);
        }

        stopwatch.Stop();
        TimeSpan elapsed = stopwatch.Elapsed;
        Console.WriteLine("FindPerson [benchmark]: " + elapsed.Milliseconds + " ms");
    }
}

File: benkchamarkphp/Person.php
<?php

class Person
{
    private $arGithubEventPrepared;
    private $arGithubEvent;

    public function __construct($id_pessoa)
    {
        $this->arGithubEventPrepared = [];
        $this->arGithubEvent = [];

        $start_time = microtime(true);
        

        $this->loadData();
        $this->prepareData();
        // $this->findPerson($id_pessoa);
        $this->searchPerson($id_pessoa);
        

        $end_time = microtime(true);
        $elapsed = $end_time - $start_time;
        echo "Total time [benchmark]: " . ($elapsed * 1000) . " ms\n"; // Display total time in milliseconds

    }

    private function loadData()
    {
        $start_time = microtime(true);
        
        $json = file_get_contents("/var/www/html/filmesApi/Controllers/TextArrayList/large-file.json");
        $this->arGithubEvent = json_decode($json, true);
        
        $end_time = microtime(true);
        $elapsed = $end_time - $start_time;
        echo "loadData [benchmark]: " . ($elapsed * 1000) . " ms \n"; // Display total time in milliseconds
    }

    private function prepareData()
    {
        $start_time = microtime(true);
        foreach ($this->arGithubEvent as $githubEvent) {
            $this->arGithubEventPrepared[$githubEvent['id']] = $githubEvent;
        }
        $end_time = microtime(true);
        $elapsed = $end_time - $start_time;
        echo "prepareData [benchmark]: " . ($elapsed * 1000) . " ms \n"; // Display total time in milliseconds
    }

    private function findPerson($id_pessoa)
    {
        $start_time = microtime(true);
        if (isset($this->arGithubEventPrepared[$id_pessoa])) {
            // print_r ($this->arGithubEventPrepared[$id_pessoa]);
        } else {
            echo "Event not found with id: " . $id_pessoa;
        }
        $end_time = microtime(true);
        $elapsed = $end_time - $start_time;
        echo "findPerson [benchmark]: " . ($elapsed * 1000) . " ms \n"; // Display total time in milliseconds
    }

    private function searchPerson($id_pessoa)
    {
        $start_time = microtime(true);
        
        foreach ($this->arGithubEvent as $githubEvent) {
            if ($githubEvent['id'] == $id_pessoa) {
                // print_r($githubEvent); // Uncomment to print the found event
                $found = true;
                break; // Exit the loop after finding the event
            }
        }

        $end_time = microtime(true);
        $elapsed = $end_time - $start_time;
        echo "findPerson [benchmark]: " . ($elapsed * 1000) . " ms \n"; // Display total time in milliseconds
    }
}

new Person('2489678844');
